core/src/main/java/org/mobilitydata/gtfsvalidator/notice/SeverityLevel.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2021 MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.notice;
+
+/**
+ * Describes the level of severity encountered during validation. 1) WARNING: the GTFS dataset lacks
+ * of quality e.g {@code UnusedTripNotice} 2) ERROR: the GTFS dataset does not comply to an explicit
+ * rule e.g. {@code MissingCalendarAndCalendarDateFilesNotice} 3) SYSTEM_ERROR: level of severity to
+ * represent an internal error
+ */
+public enum SeverityLevel {
+  WARNING,
+
+  ERROR,
+
+  SYSTEM_ERROR;
```

Comment:
I suggest the following constants:

* INFO - for things that do not affect feed quality, such as unknown files or unknown fields;
* WARNING - affect feed quality (e.g., insufficient color contrast) but the feed remains functional;
* ERROR - the feed cannot function (e.g., broken foreign key reference);

We may also add CRITICAL since many logging frameworks have that concept but I do not see how that differs from ERROR in our case.

System errors are all ERRORs by definition. A system error is an internal error in the validator and this means that the feed was not processed successfully.

Reply:
Please leave comments at each of the constants.